android/src/com/mapswithme/util/Utils.java

```diff
@@ -289,4 +289,10 @@ public static void navigateToParent(Activity activity)
     else
       NavUtils.navigateUpFromSameTask(activity);
   }
+
+  public static boolean directoryExists(String path)
+  {
+    final File file = new File(path);
+    return (file.exists() && file.isDirectory() && file.canRead());
```

Comment:
Method name diverges from it's semantics. The check should be changed or the method name to "directoryExistsAndCanBeRead".